### ISSUE TITLE
Bug: Can't rewatermark file

### ISSUE THREAD

**Issue by NotSqrt** [pmaupin/pdfrw#50]
Hi,

Just discovered a small bug:

code to reproduce:

``` python
from reportlab.pdfgen import canvas
from pdfrw import PdfReader, PdfWriter, PageMerge


# create some files
pdf_file = canvas.Canvas('page.pdf')
pdf_file.drawString(0, 0, 'hello')
pdf_file.save()

watermark_file = canvas.Canvas('water.pdf')
watermark_file.drawString(0, 0, 'water')
watermark_file.save()


# watermark 1
wmark = PageMerge().add(PdfReader('water.pdf').pages[0])[0]
trailer = PdfReader('page.pdf')

for page in trailer.pages:
    PageMerge(page).add(wmark).render()

PdfWriter().write('merged.pdf', trailer)


# watermark the watermarked file
trailer = PdfReader('merged.pdf')

for page in trailer.pages:
    PageMerge(page).add(wmark).render()

PdfWriter().write('merged2.pdf', trailer)
```

The problem is around https://github.com/pmaupin/pdfrw/blob/master/pdfrw/pagemerge.py#L202.

The number 6 is not `len('\pdfrw_')`, so the `isdigit()` fails, and `\pdfrw_0` is re-used every time.

**Comment by cristianocca** [comment]
The same happens if it is scaled or what ever that uses render at the end. Any news on this?

**Comment by cristianocca** [comment]
Looks like it can be fixed changing the following line:
https://github.com/pmaupin/pdfrw/blob/master/pdfrw/pagemerge.py#L205

from `key_offset -= len(allkeys)` to `key_offset += len(allkeys)`

Not really sure about the implications of this, but water marking again works and no crash. Any chance this can be confirmed and pushed?

**Comment by cristianocca** [comment]
FYI I think this was fixed on some patch since it's no longer an issue with pypi's 3.0 version.

**Comment by pmaupin** [comment]
Yes, this was fixed by PR #56 -- closing.
